presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkServiceFactory.java

```diff
@@ -47,17 +47,17 @@ public IPrestoSparkService createService(SparkProcessType sparkProcessType, Pres
                 configuration.getSessionPropertyConfigurationProperties(),
                 configuration.getFunctionNamespaceProperties(),
                 getSqlParserOptions(),
-                getAdditionalModules());
+                getAdditionalModules(configuration));
 
         Injector injector = prestoSparkInjectorFactory.create();
         PrestoSparkService prestoSparkService = injector.getInstance(PrestoSparkService.class);
         log.info("Initialized");
         return prestoSparkService;
     }
 
-    protected List<Module> getAdditionalModules()
+    protected List<Module> getAdditionalModules(PrestoSparkConfiguration configuration)
     {
-        return ImmutableList.of();
+        return ImmutableList.of(new PrestoSparkMetadataStorageModule());
```

Comment:
let's check `checkArgument("local".equals(configuration.getMetadataStorage()), "only local metadata storage is supported")`